src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -3251,11 +3255,11 @@ def aks_agentpool_update(cmd,   # pylint: disable=unused-argument
     update_autoscaler = enable_cluster_autoscaler + \
         disable_cluster_autoscaler + update_cluster_autoscaler
 
-    if (update_autoscaler != 1 and not tags and not mode and not max_surge):
+    if (update_autoscaler != 1 and not tags and not scale_down_mode and not mode and not max_surge):
```

Comment:
this if worth some unit test.

Reply:
since we have no unit test setup here. I think it's ready to go.
Fuming will setup the unit test structure so people can add ut later.